src/app/Shared/Services/Api.service.tsx

```diff
@@ -476,6 +476,81 @@ export class ApiService {
     );
   }
 
+  removeProbes(): Observable<boolean> {
+    return this.target.target().pipe(
+      concatMap((target) =>
+        this.sendRequest('v2', `targets/${encodeURIComponent(target.connectUrl)}/probes`, {
+          method: 'DELETE',
+        }).pipe(
+          tap((resp) => {
+            if (resp.status == 400) {
+              this.notifications.warning('Failed to remove Probes', 'The probes failed to be removed from the target');
+            }
+          }),
+          map((resp) => resp.status == 200),
+          first()
+        )
+      )
+    );
+  }
+
+  insertProbes(templateName: string): Observable<boolean> {
+    return this.target.target().pipe(
+      concatMap((target) =>
+        this.sendRequest(
+          'v2',
+          `targets/${encodeURIComponent(target.connectUrl)}/probes/${encodeURIComponent(templateName)}`,
+          {
+            method: 'POST',
+          }
+        ).pipe(
+          tap((resp) => {
+            if (resp.status == 400) {
+              this.notifications.warning(
+                'Failed to Insert Probes',
+                'The probes failed to be injected. Check that the agent is present in the same container as the target JVM and the target is running with -javaagent:/path/to/agent'
+              );
+            }
+          }),
+          map((resp) => resp.status == 200),
+          first()
+        )
+      )
+    );
+  }
+
+  addCustomProbeTemplate(file: File): Observable<boolean> {
+    const body = new window.FormData();
+    body.append('probeTemplate', file);
+    return this.sendRequest('v2', `probes/` + file.name, {
```

Comment:
Not a web issue but just a question for this endpoint: Any reason that backend requires the file name to be in path?

Reply:
That endpoint uses the filename as the resource name.

```xml
<?xml version="1.0"?>
<jfragent>
  <!-- Global configuration options -->
  <config>
    <!-- This is the prefix to use when generating event class names. -->
    <classprefix>__JFREvent</classprefix>
    <!-- Will allow the recording of arrays and object parameters as Strings. This will cause toString to be called for array elements and objects other than strings, which in turn can cause trouble if the toString method is badly implemented. Use with care. -->
    <allowtostring>true</allowtostring>
    <!-- Allows converters to be used. See the org.openjdk.jmc.agent.converters package for more information. -->
    <allowconverter>true</allowconverter>
  </config>
  <events>
    <event id="demo.jfr.test1">
      <label>DemoEvent</label>
      <description>Event for the agent plugin demo</description>
      <path>demo</path>
      <stacktrace>true</stacktrace>
      <class>io.cryostat.net.web.http.api.v1.TargetTemplatesGetHandler</class>
      <method>
        <name>handleAuthenticated</name>
        <descriptor>(Lio/vertx/ext/web/RoutingContext;)V</descriptor>
      </method>
      <!-- location {ENTRY, EXIT, WRAP}-->
      <location>ENTRY</location>
    </event>
  </events>
</jfragent>
```

This is an example of what the XML resource looks like - it doesn't have a meaningful name element to extract, or anything like that. A future Cryostat bcakend version should probably register these in a database and assign them numeric IDs instead.

Reply:
Right! That makes sense! Thanks for explaining :D